indexer-app/src/cmr/indexer/data/concepts/keyword_util.clj

```diff
@@ -34,13 +42,71 @@
 
 (defn- contact-person->keywords
   "Converts a compound field into a vector of terms for keyword searches."
-  [data]
+  [contact-person]
   (let [{first-name :FirstName
          last-name :LastName
-         roles :Roles} data]
+         roles :Roles} contact-person]
     (concat [first-name last-name]
             roles)))
 
+(defn- get-contact-persons
+  "Retrieve a vector of contact persons."
+  [data]
+  (let [{:keys [ContactPersons ContactGroups DataCenters]} data]
+    (concat ContactPersons ContactGroups
+     (mapcat :ContactGroups DataCenters)
+     (mapcat :ContactPersons DataCenters))))
+
+(defn- get-contact-mechanisms->keywords
+  "Retrieve contact mechanisms and convert into a vector of terms for keyword searches."
+  [data]
+  (map #(:Value (first %))
+       (map #(get-in % [:ContactInformation :ContactMechanisms])
+            (get-contact-persons data))))
+
+(defn- data-center->keywords
+  "Convert a compound field into a vector of terms for keyword searches."
+  [data-center]
+  (let [{contact-persons :ContactPersons
+         contact-groups :ContactGroups} data-center]
+    (concat (mapcat contact-person->keywords contact-persons)
+            (mapcat contact-group->keywords contact-groups)
+            [(:ShortName data-center)])))
+
+(defn- collection-citation->keywords
+  "Convert a compound field into a vector of terms for keyword searches."
+  [collection-citation]
+  [(:Creator collection-citation)
+   (:OtherCitationDetails collection-citation)])
+
+(defn- characteristic->keywords
+  "Convert a compound field into a vector of terms for keyword searches."
+  [characteristic]
+  [(:name characteristic)
+   (:description characteristic)
+   (:value characteristic)])
+
+(defn- collection-platforms->keywords
+  "Convert a collection of platforms into a vector of terms for keyword searches."
```

Comment:
The returned value of this function is a list of terms, not a vector. We should update the docstring of the function to something like "Convert the given platforms to a list of terms for keyword searches."